src/index.js

```diff
@@ -44,14 +48,56 @@ export async function create ({
       const { method, url, headers } = req
       debug('Request: %O', { method, url, headers })
 
-      // TODO: Show something at the root?
-      if (!url.startsWith('/hyper/')) {
-        res.writeHead(404)
-        res.end('Not Found')
-        return
+      let finalURL = null
+      let isSubdomained = false
+
+      if (subdomain || subdomainRedirect) {
+        const { host } = headers
+        if (host) {
+          const segments = host.split('.')
+          const [subdomain] = segments
+          if (segments.length > 2) {
+            isSubdomained = true
+
+            if (subdomain.length === 52) {
+              // Likely a hypercore key
+              finalURL = HYPER_PREFIX + subdomain + url
+            } else if (subdomain.includes('-')) {
+            // Likely an escaped subdomain
+              const unescaped = subdomain
+                .split('--')
+                .map((section) => section.replaceAll('-', '.'))
+                .join('-')
+
+              finalURL = HYPER_PREFIX + unescaped + url
+            }
+          }
+        }
       }
 
-      const finalURL = 'hyper://' + url.slice('/hyper/'.length)
+      if (!finalURL) {
+        // TODO: Show something at the root?
+        if (!url.startsWith('/hyper/')) {
+          res.writeHead(404)
+          res.end('Not Found')
+          return
+        }
+
+        finalURL = HYPER_PREFIX + url.slice('/hyper/'.length)
+
+        if (subdomainRedirect && !isSubdomained) {
+        // Take the host from the URL and redirect to the subdomain
+          const host = headers.host
+          const [hyperHost, ...pathSegments] = finalURL.slice(HYPER_PREFIX.length).split('/')
+          const subdomainName = hyperHost.replaceAll('-', '--').replaceAll('.', '-')
```

Comment:
maybe we should have a function for escaping and unescaping domains instead of repeating the logic